src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1513,6 +1514,57 @@ public string GetMemberQueryPath(Expression expression)
             return propertyName;
         }
 
+        internal void VectorSearch(VectorEmbeddingFieldFactory<T> embeddingFieldFactory, VectorFieldValueFactory embeddingValueFactory,
+            float minimumSimilarity, int numberOfCandidates, bool isExact)
+        {
+            var fieldName = embeddingFieldFactory.FieldName;
+            var sourceQuantizationType = embeddingFieldFactory.SourceQuantizationType;
+            var targetQuantizationType = embeddingFieldFactory.DestinationQuantizationType;
+            var isSourceBase64Encoded = embeddingFieldFactory.IsBase64Encoded;
+            
+            string queryParameterName;
+            var isVectorBase64Encoded = false;
+            
+            if (embeddingValueFactory.Text != null)
+            {
+                queryParameterName = AddQueryParameter(embeddingValueFactory.Text);
+            }
+
+            else if (embeddingValueFactory.Embedding != null)
+            {
+                // for well-known types we can convert the array into Base64
+                queryParameterName = AddQueryParameter(embeddingValueFactory.Embedding switch
+                {
+                    float[] fa => Convert.ToBase64String(MemoryMarshal.Cast<float, byte>(fa)
+#if !NETCOREAPP3_1_OR_GREATER
+                        .ToArray() // For newer frameworks we use Span overload in Convert.
+#endif
+                    ),
+                    byte[] ba => Convert.ToBase64String(ba),
+                    sbyte[] sb => Convert.ToBase64String(MemoryMarshal.Cast<sbyte, byte>(sb)
+#if !NETCOREAPP3_1_OR_GREATER
+                        .ToArray()
+#endif
+                    ),
+                    _  => embeddingValueFactory.Embedding,
```

Comment:
TODO benchmark

Currently, parsing a float array from a blittable array is expensive, so we could just send the base64-encoded array. However, once https://github.com/ravendb/ravendb/pull/19603 is completed, we can send the floats in vector array form.